### PR TITLE
Make a smaller param for ard, and fix order of params

### PR DESCRIPTION
**Pull request by bradfrizzell**
We had been passing the entire url that we call out to, now instead we shorten it. So instead of a doubleclick request including, for example:
...&ard=https://www.example.com/endpoint.php?a=1&b=2&c=3&d=4
it would now include 
 &ard=www.example.com/endpoint.php

### CODE REVIEW

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 1]
```diff
@@ -66,13 +69,31 @@ export const RTC_ERROR_ENUM = {
...
+ * passed "https://example.com/example.php?foo=a&bar=b, then we return
+ * example.com/example.php
+ * and if we are passed "exampleVendor" then we return "exampleVendor"
+ * @param {string} callout
+ * @return {string}
+ */
+export function getCalloutParam(callout) {
+  try {
+    const parsedUrl = parseUrl(callout);
+    callout = (parsedUrl.hostname + parsedUrl.pathname).substr(0, 50);
```
Comment (jeffkaufman) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: The 50 character limit should be doc'd

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 2]
```diff
@@ -66,13 +69,31 @@ export const RTC_ERROR_ENUM = {
...
+  } catch (e) {}
```
Comment (jeffkaufman) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: What sort of errors are we trying to catch here?

We should generally be doc'ing what sort of errors lead to us just returning the input unmodified.

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 3]
```diff
@@ -225,7 +246,8 @@ function sendRtcCallout_(
             return {rtcTime, callout};
```
Comment (jeffkaufman) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: should `callout` here be trimmed?

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 4]
```diff
@@ -225,7 +246,8 @@ function sendRtcCallout_(
...
           }
           const response = tryParseJson(text);
-          return response ? {response, rtcTime, callout} :
+          return response ? {response, rtcTime,
+            callout: getCalloutParam(callout)} :
             buildErrorResponse_(
                 RTC_ERROR_ENUM.MALFORMED_JSON_RESPONSE, callout, rtcTime);
```
Comment (jeffkaufman) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: what about `callout` here?

File: extensions/amp-a4a/0.1/test/test-real-time-config-manager.js [hunk 5]
```diff
@@ -376,9 +381,12 @@ describes.realWin('real-time-config-manager', {amp: true}, env => {
...
-        {callout: urls[2], error: RTC_ERROR_ENUM.INSECURE_URL, rtcTime: 10},
+        {response: rtcCalloutResponses[0],
+          callout: getCalloutParam(urls[0]), rtcTime: 10},
+        {response: rtcCalloutResponses[1],
+          callout: getCalloutParam(urls[1]), rtcTime: 10},
+        {callout: getCalloutParam(urls[2]),
+          error: RTC_ERROR_ENUM.INSECURE_URL, rtcTime: 10},
       ];
       return executeTest({
         urls, inflatedUrls: urls, rtcCalloutResponses, calloutCount,
```
Comment (jeffkaufman) [inline extensions/amp-a4a/0.1/test/test-real-time-config-manager.js]: You also need explicit tests of `getCalloutParam`: I think these tests would all pass if it just always returned `foo`.

File: extensions/amp-a4a/0.1/real-time-config-manager.js (hunk 3 above)
Comment (bradfrizzell) [reply extensions/amp-a4a/0.1/real-time-config-manager.js]: Changed where we do the trimming so we only need to do it once.

File: extensions/amp-a4a/0.1/real-time-config-manager.js (hunk 2 above)
Comment (bradfrizzell) [reply extensions/amp-a4a/0.1/real-time-config-manager.js]: done

File: extensions/amp-a4a/0.1/real-time-config-manager.js (hunk 1 above)
Comment (bradfrizzell) [reply extensions/amp-a4a/0.1/real-time-config-manager.js]: done

File: extensions/amp-a4a/0.1/test/test-real-time-config-manager.js [hunk 6]
```diff
@@ -138,8 +154,10 @@ describes.realWin('real-time-config-manager', {amp: true}, env => {
       return urls;
     }
 
-    function rtcEntry(response, url, error) {
-      const callout = getCalloutParam(url);
+    function rtcEntry(response, url, error, isVendor) {
+      // If this is an entry for a vendor, then the callout is just
+      // the vendor name passed in to url here.
+      const callout = !!isVendor ? url : getCalloutParam_(url);
```
Comment (jeffkaufman) [inline extensions/amp-a4a/0.1/test/test-real-time-config-manager.js]: minor: is that `!!` actually doing anything?

Comment (bradfrizzell) [reply extensions/amp-a4a/0.1/test/test-real-time-config-manager.js]: I suppose not, I just put it there for sanity sometimes as I'm going

File: extensions/amp-a4a/0.1/real-time-config-manager.js [hunk 7]
```diff
@@ -152,22 +171,22 @@ export function inflateAndSendRtc_(a4aElement, url, seenUrls, promiseArray,
     if (Object.keys(seenUrls).length == MAX_RTC_CALLOUTS) {
       return buildErrorResponse_(
           RTC_ERROR_ENUM.MAX_CALLOUTS_EXCEEDED,
-          opt_vendor || url, undefined, true);
+          callout, undefined, true);
```
Comment (keithwrightbos) [inline extensions/amp-a4a/0.1/real-time-config-manager.js]: Let's always report errors... the only place where I don't see that we do is for malformed json and I can't see why we wouldn't.  Just remove opt_log and then you can remove all of these undefined portions for opt_rtcTime

**Pull request merged**